File: Modules/ServerTypes/PerformanceAppraisal/PerformanceAppraisalFileAttachForm.ts
import { ImageUploadEditor, StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PerformanceAppraisalFileAttachForm {
    Files: ImageUploadEditor;
    Remark: StringEditor;
}

export class PerformanceAppraisalFileAttachForm extends PrefixedContext {
    static formKey = 'PerformanceAppraisal.PerformanceAppraisalFileAttach';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PerformanceAppraisalFileAttachForm.init)  {
            PerformanceAppraisalFileAttachForm.init = true;

            var w0 = ImageUploadEditor;
            var w1 = StringEditor;

            initFormType(PerformanceAppraisalFileAttachForm, [
                'Files', w0,
                'Remark', w1
            ]);
        }
    }
}

File: Modules/HumanResource/PerformanceAppraisal/PerformanceAppraisalFileAttach/PerformanceAppraisalFileAttachForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.PerformanceAppraisal.Forms;

[FormScript("PerformanceAppraisal.PerformanceAppraisalFileAttach")]
[BasedOnRow(typeof(PerformanceAppraisalFileAttachRow), CheckNames = true)]
public class PerformanceAppraisalFileAttachForm
{
    public string Files { get; set; }
    public string Remark { get; set; }
}